Source/WebKit/UIProcess/WebPageProxy.h

```diff
@@ -2917,10 +2918,39 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
     bool m_isListeningForUserFacingStateChangeNotification { false };
 #endif
     bool m_allowsMediaDocumentInlinePlayback { false };
-    std::unique_ptr<ProcessThrottlerActivity> m_isVisibleActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_isAudibleActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_isCapturingActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_openingAppLinkActivity;
+
+    struct ProcessActivityState {
+        WTF_MAKE_STRUCT_FAST_ALLOCATED;
+
+        WebPageProxy& m_page;
+
+        std::unique_ptr<ProcessThrottlerActivity> m_isVisibleActivity;
+        UniqueRef<ProcessThrottlerTimedActivity> m_wasRecentlyVisibleActivity;
```

Comment:
I think we should guard this behind `PLATFORM(MAC)` since this is so macOS-specific.